[WIP] Add source content

zeke opened the pull request:
This PR is a first step toward collecting source content from the nodejs/node repo to be synced with Crowdin. The `collect` script uses semver.io to determine the "Current" version of Node.js, but we should discuss which versions we want to collect, and how to structure them in this repo.

cc @nodejs/i18n 

🍐 with @vanessayuenn

obensource commented:
@zeke @vanessayuenn holy smokes! This is amazing!!! Thank you for starting us off on such a great foot! Can't wait to get this synced with Crowdin asap! 🙌

zeke commented:
Note that as we add new versions of node's docs (e.g 9.9.1 when that comes out) we can configure [a Crowdin workflow that applies all existing translations from the Translation Memory](https://github.com/electron/i18n/issues/135). This way translators will only need to translate content that has changed between version.

zeke commented:
See also this Electron issue where we ponder what Electron versions to localize: https://github.com/electron/i18n/issues/288

zeke commented:
I made a few updates:

- Updated the `collect` script to: 
  - support fetching multiple versions
  - skip versions that have already been downloaded
  - only keep `.md` files, ignoring images and any other files
- Took the fetched docs out of this PR for now, to make reviewing easier
- Added an explicit list of target Node.js versions for starters (LTS and Current)
- Added linting and tests

zeke commented:
Taking `WIP` off the title. This is ready for review.

lukaszewczak commented:
Hi @zeke , I try to run this script (on Windows), but I receive 404 (Not Found) because of the wrong url

```
downloading https://github.com/nodejs/node/archive/vv8.11.0.tar.gz
(node:17156) UnhandledPromiseRejectionWarning: HTTPError: Response code 404 (Not Found)
```

If I remove `v` from node.js versions in `package.json` in section `nodeVersions` it runs.

```
 "nodeVersions": [
    "8.11.0",
    "9.10.0"
  ],
```

But then I receive this error during moving docs from temp dir, it probably should by copied to the destination folder

```
downloading https://github.com/nodejs/node/archive/v8.11.0.tar.gz
(node:17100) UnhandledPromiseRejectionWarning: Error: EXDEV: cross-device link not permitted, rename 'C:\Users\Dom\AppData\Local\Temp\d-118229-17100-1103929.rp0z\node-8.11.0\doc' -> 'D:
\Projekty\Learn\i18n\content\v8.11.0\en-US\doc'
    at Object.fs.renameSync (fs.js:788:18)
    at getDocsForNodeVersion (D:\Projekty\Learn\i18n\script\collect.js:37:6)
    at <anonymous>

```

FranzDeCopenhague commented on script/collect.js:
```diff
@@ -0,0 +1,43 @@
+}
+
+async function getDocsForNodeVersion (version) {
+  const docDir = path.join(__dirname, `../content/v${version}/en-US/doc`)
```
@lukaszewczak mentioned that package.json already has the "v" prefix at 
```
 "nodeVersions": [
   "v8.11.0",
   "v9.10.0"
  ]
```

FranzDeCopenhague commented on script/collect.js:
```diff
@@ -0,0 +1,43 @@
+  }
+
+  // download repo bundle and extract to a temporary directory
+  const tarballUrl = `https://github.com/nodejs/node/archive/v${version}.tar.gz`
```
@lukaszewczak mentioned that package.json already has the "v" prefix at 
```
 "nodeVersions": [
   "v8.11.0",
   "v9.10.0"
  ]
```

FranzDeCopenhague commented on test.js:
```diff
@@ -0,0 +1,37 @@
+test('includes source English content for all nodeVersions', () => {
+  const fetchedVersions = fs.readdirSync(contentDir)
+  expect(nodeVersions).toEqual(fetchedVersions)
+})
```
The test title refers to English content, I suggest checking the inner language folders
```
fetchedVersions.forEach(version => {
    const languages = fs.readdirSync(path.join(contentDir, version))
    expect(languages).include('en-US')
})

zeke commented:
Thanks for giving it a go, @lukaszewczak and @FranzDeCopenhague. Should be fixed now. Can you give it another try?

I was waffling between having the `v` prefix everywhere, and decided to keep it because it's generally safer for variables, filenames, etc to start with a character than a number. Also `semver` gracefully tolerates a `v` prefix:

```js
semver.valid('v1.2.3') // true
```

zeke commented:
Thanks @FranzDeCopenhague, though I kind of wanted to get more feedback from other reviewers, but I suppose we can iterate from here in new pull requests.

I opened https://github.com/nodejs/i18n/issues/59 to open the discussion about how we can enforce review requirements.

FranzDeCopenhague commented:
@zeke, I did wait,  a little, for @lukaszewczak feedback and move on because the issue looks like a windows problem where the temp folder is at a different unit drive to the i18n project

zeke commented:
> But then I receive this error during moving docs from temp dir, it probably should by copied to the destination folder

This sounds familiar. See https://github.com/electron-userland/electron-download/pull/36 -- I agree that we should download to the repo directory.

lukaszewczak commented:
Hi, I'm sorry but I wasn't able to look at your changes yesterday @zeke. Download directly to the repo folder sounds good and should make script platform independent. Will it be ok for you @zeke, if try to prepare PR with this change?

zeke commented:
> Will it be ok for you @zeke, if try to prepare PR with this change?

Of course! Did you mean me or you, though? :)

lukaszewczak commented:
Me :-) I will try to prepare it:-)
